### PR TITLE
Allow switching to daily instead of hourly index buckets

### PR DESCRIPTION
**Pull request by juliusv**
There can be slight overlap around the switchover time, but I deemed it
safer (and not incorrect) than risking gaps due to some calculation
error.

### CODE REVIEW

**Comment by juliusv** [comment]
@tomwilkie This would be the idea - haven't tested it fully on Kubernetes yet, only the unit test, but will do when I get it all running again.

File: chunk/chunk_store.go [hunk 1]
```diff
@@ -332,25 +338,45 @@ func (c *AWSStore) CreateTables() error {
 	return err
 }
 
-func bigBuckets(from, through model.Time) []int64 {
+func bigBuckets(from, through model.Time, dailyBucketsFrom model.Time) []string {
```
Comment (tomwilkie) [inline chunk/chunk_store.go]: I'd make this a method on `AWSStore` and not pass `dailyBucketsFrom` as a param.

Comment (juliusv) [inline chunk/chunk_store.go]: Done.

File: chunk/chunk_store.go [hunk 2]
```diff
@@ -332,25 +338,45 @@ func (c *AWSStore) CreateTables() error {
...
+
+		result []string
 	)
+
 	for i := fromHour; i <= throughHour; i++ {
-		result = append(result, i)
+		if i > lastHourlyBucket {
+			break
+		}
+		result = append(result, strconv.Itoa(int(i)))
```
Comment (tomwilkie) [inline chunk/chunk_store.go]: Not sure the conversion to int is safe here - they'll run out on 2038-01-19 IIRC.  Should probably use https://golang.org/pkg/strconv/#FormatInt

**Comment by tomwilkie** [comment]
Hmm, looks like leap second handling in unix time isn't what I thought it would be (surprise!): https://en.wikipedia.org/wiki/Unix_time#Leap_seconds.  Unix time is contiguous.  Given that we receive the samples with a unix-like ms timestamp from the client, not sure there is much we can do here.

File: chunk/chunk_store.go (hunk 2 above)
Comment (tomwilkie) [inline chunk/chunk_store.go]: Ah ignore me, these are hours, so the range is huge.

**Comment by tomwilkie** [comment]
I'm pretty happy with this, so going to merge.  I'm touching the same code as you, so this will save a bad merge.

File: chunk/chunk_store.go [hunk 3]
```diff
@@ -206,6 +208,10 @@ type AWSStore struct {
 	tableName  string
 	bucketName string
 
+	// Around which time to start bucketing indexes by day instead of by hour.
+	// Only the day matters, not the time within the day.
+	dailyBucketsFrom model.Time
```
Comment (tomwilkie) [inline chunk/chunk_store.go]: This variable is never initialised!  Which means we'll default to weekly buckets for everything...

We should do two things: copy the config straight into the AWSStore struct, and change the meaning so the time-zero-value means no weekly buckets.

Will backout this PR.

**Comment by juliusv** [comment]
Sorry about that - thus the remark that I hadn't tried it out yet ;) Hope it didn't cause too much havok (only auto-rolled-out to dev after merging, I assume).

**Comment by tomwilkie** [comment]
Yeah no worries, as much my bad for not catching / testing it myself before merging.

**Pull request merged**